relayer/pkg/chainlink/txm/txm.go

```diff
@@ -125,7 +197,82 @@ func (txm *starktxm) broadcastLoop() {
 	}
 }
 
+func (txm *starktxm) handleBroadcastErr(ctx context.Context, data any, accountAddress *felt.Felt, publicKey *felt.Felt, call starknetrpc.FunctionCall) error {
+
+	errData := fmt.Sprintf("%s", data)
+	txm.lggr.Debug("encountered handleBroadcastErr", errData)
+
+	if isInvalidNonce(errData) {
+		// resubmits all unconfirmed transactions
+		err := txm.handleNonceErr(ctx, accountAddress, publicKey)
+		if err != nil {
+			return pkgerrors.Wrap(err, "error in nonce handling")
+		}
+		// resubmits the current 1 unbroadcasted tx that just failed
+		err = txm.Enqueue(accountAddress, publicKey, call)
+		if err != nil {
+			return pkgerrors.Wrap(err, "error in re-enqueuing after nonce handling")
+		}
+	}
+
+	return nil
+}
+
+func (txm *starktxm) handleNonceErr(ctx context.Context, accountAddress *felt.Felt, publicKey *felt.Felt) error {
+
+	txm.lggr.Debugw("Handling Nonce Validation Error By Resubmitting Txs...", "account", accountAddress)
+
+	// exponential backoff wait for rpc starknet_estimateFee to catch up with nonce returned by starknet_getNonce
+	txm.nonceRetryTimer.Wait(ctx)
```

Comment:
look at avoiding waiting for nonce to resolve in both confirmer and broadcaster.